lisa/analysis/rta.py

```diff
@@ -15,187 +15,493 @@
 # limitations under the License.
 #
 
-import os
-import re
-import glob
+from collections import namedtuple
 
-import pandas as pd
+from lisa.analysis.base import AnalysisHelpers, TraceAnalysisBase
+from lisa.datautils import df_filter_task_ids
+from lisa.trace import TaskID, requires_events
+from lisa.utils import memoized
 
-from lisa.utils import Loggable
-from lisa.analysis.base import AnalysisHelpers
-
-class PerfAnalysis(AnalysisHelpers):
+class RTAEventsAnalysis(TraceAnalysisBase):
     """
-    Parse and analyse a set of RTApp log files
-
-    :param task_log_map: Mapping of task names to log files
-    :type task_log_map: dict
+    Support for RTA events analysis.
 
-    .. note:: That is not a subclass of
-        :class:`lisa.analysis.base.TraceAnalysisBase` since it does not uses traces.
+    :param trace: input Trace object
+    :type trace: lisa.trace.Trace
     """
 
     name = 'rta'
 
-    RTA_LOG_PATTERN = 'rt-app-{task}.log'
-    "Filename pattern matching RTApp log files"
-
-    def __init__(self, task_log_map):
+    def __init__(self, trace):
```

Comment:
Since it no longer has any specific content other than calling `super().__init__`, `__init__` can be dropped completely

Reply:
Yep, indeed... :/